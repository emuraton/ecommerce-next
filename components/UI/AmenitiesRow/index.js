import React from 'react';
import styled from 'styled-components';

import { Wifi, Parking, Bar, AC } from '../SVG/Amenities';

const List = styled.ul`
  list-style: none;
  padding: 0;
  display: flex;
  flex-wrap: wrap;
`;

const ListItem = styled.li`
  position: relative;
  color: #37454d;
  overflow: visible;
  text-align: center;
  width: 33%;
  /* height: 52px; */
  float: left;
  text-align: left;
  /* padding-right: 16px; */
`;
ListItem.displayName = 'ListItem';

const Icon = styled.span`
  display: inline-block;
  position: relative;
  width: 24px;
  height: 24px;
  vertical-align: middle;
  margin: 0 12px 0 0;
`;

const IconLabel = styled.span`
  display: inline-block;
  overflow: hidden;
  vertical-align: middle;
  font-weight: 500;
  color: rgb(72, 72, 72);
  font-size: 16px;
  line-height: 25px;
`;

const Container = styled.div`
  margin-top: 10px;
  border-top: 1px solid rgb(235, 235, 235);
`;

const ShowMoreButton = styled.button`
  background: transparent;
  color: rgb(0, 132, 137);
  border: none;
  text-decoration: none;
  font-size: 15px;
  line-height: 18px;
  font-weight: 500;
  cursor: pointer;
  outline: none;
`;
ShowMoreButton.displayName = 'ShowMoreButton';

export const renderAmenity = slug => enumAmenities[slug] || <Wifi />;

const AmenitiesRow = ({ amenities }) => {
  if (!amenities || amenities.length === 0) return null;
  return (
    <Container>
      <List>
        {amenities.map((amenity, index) => {
          if (index > 2) return null; // Only display first 3 amenities
          return (
            <ListItem key={`amentity-${index}`}>
              <Icon>{renderAmenity(amenity.slug)}</Icon>
              <IconLabel>{amenity.name}</IconLabel>
            </ListItem>
          );
        })}
      </List>
      {amenities.length > 2 && (
        <ShowMoreButton type="button" onClick={() => {}}>
          + Show all amenities
        </ShowMoreButton>
      )}
    </Container>
  );
};

const enumAmenities = {
  wifi: <Wifi />,
  bar: <Bar />,
  AC: <AC />,
  parking: <Parking />,
};

export default AmenitiesRow;
