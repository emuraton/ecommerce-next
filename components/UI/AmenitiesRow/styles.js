import styled from 'styled-components';

const List = styled.ul`
  list-style: none;
  padding: 0;
  display: flex;
  flex-wrap: wrap;
`;

const ListItem = styled.li`
  position: relative;
  color: #37454d;
  overflow: visible;
  text-align: center;
  width: 33%;
  float: left;
  text-align: left;
`;
ListItem.displayName = 'ListItem';

const Icon = styled.span`
  display: inline-block;
  position: relative;
  width: 24px;
  height: 24px;
  vertical-align: middle;
  margin: 0 12px 0 0;
`;

const IconLabel = styled.span`
  display: inline-block;
  overflow: hidden;
  vertical-align: middle;
  font-weight: 500;
  color: rgb(72, 72, 72);
  font-size: 16px;
  line-height: 25px;
`;

const Container = styled.div`
  margin-top: 10px;
  border-top: 1px solid rgb(235, 235, 235);
  border-bottom: 1px solid rgb(235, 235, 235);
`;

const ShowMoreButton = styled.button`
  background: transparent;
  color: rgb(0, 132, 137);
  border: none;
  text-decoration: none;
  font-size: 15px;
  line-height: 18px;
  font-weight: 500;
  cursor: pointer;
  outline: none;
`;
ShowMoreButton.displayName = 'ShowMoreButton';

const Amenities = {
  Container,
  ShowMoreButton,
  Icon,
  IconLabel,
  List,
  ListItem,
};

export default Amenities;
