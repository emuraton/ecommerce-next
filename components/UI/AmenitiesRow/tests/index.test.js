import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import AmenitiesRow from '..';

describe('<AmenitiesRow />', () => {
  describe('@renders', () => {
    it('default render', () => {
      const wrapper = shallow(<AmenitiesRow />);
      expect(toJson(wrapper)).toBe('');
    });

    it('with 3 amenities', () => {
      const amenities = [
        { name: 'Wifi', slug: 'wifi' },
        { name: 'Bar', slug: 'bar' },
        { name: 'A/C', slug: 'AC' },
      ];
      const wrapper = shallow(<AmenitiesRow amenities={amenities} />);
      const listItem = wrapper.find('ListItem');

      expect(toJson(wrapper)).toMatchSnapshot();
      expect(listItem).toHaveLength(3);
    });

    it('with more than 3 amenities', () => {
      const amenities = [
        { name: 'Wifi', slug: 'wifi' },
        { name: 'Bar', slug: 'bar' },
        { name: 'A/C', slug: 'AC' },
        { name: 'Parking', slug: 'parking' },
      ];
      const wrapper = shallow(<AmenitiesRow amenities={amenities} />);

      const listItem = wrapper.find('ListItem');
      expect(listItem).toHaveLength(3);
      const showMoreButton = wrapper.find('ShowMoreButton');
      expect(toJson(showMoreButton)).toMatchSnapshot();
    });
  });
});
