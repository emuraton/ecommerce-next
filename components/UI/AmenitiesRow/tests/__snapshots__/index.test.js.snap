// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AmenitiesRow /> @renders with 3 amenities 1`] = `
<AmenitiesRow__Container>
  <AmenitiesRow__List>
    <ListItem
      key="amentity-0"
    >
      <AmenitiesRow__Icon>
        <Wifi />
      </AmenitiesRow__Icon>
      <AmenitiesRow__IconLabel>
        Wifi
      </AmenitiesRow__IconLabel>
    </ListItem>
    <ListItem
      key="amentity-1"
    >
      <AmenitiesRow__Icon>
        <Bar />
      </AmenitiesRow__Icon>
      <AmenitiesRow__IconLabel>
        Bar
      </AmenitiesRow__IconLabel>
    </ListItem>
    <ListItem
      key="amentity-2"
    >
      <AmenitiesRow__Icon>
        <AC />
      </AmenitiesRow__Icon>
      <AmenitiesRow__IconLabel>
        A/C
      </AmenitiesRow__IconLabel>
    </ListItem>
  </AmenitiesRow__List>
  <ShowMoreButton
    onClick={[Function]}
    type="button"
  >
    + Show all amenities
  </ShowMoreButton>
</AmenitiesRow__Container>
`;

exports[`<AmenitiesRow /> @renders with more than 3 amenities 1`] = `
<ShowMoreButton
  onClick={[Function]}
  type="button"
>
  + Show all amenities
</ShowMoreButton>
`;
